### PR TITLE
Updated isHomed [OpenPnP formatted]

### PR DESCRIPTION
**Pull request by mgrl**
# Description
as mentioned in #804 I added an ishomed flag

# Implementation Details
1. How did you test the change? Be descriptive. Untested code will not be accepted.
live on machine and mvn
2. Did you follow the [coding style](https://github.com/openpnp/openpnp/wiki/Developers-Guide#coding-style)?
unfortunately I had to reformat the file due to some whitespace issues... I will fix the curly brakets later.

I felt it would be better to have the isHomed flag at the machine, right? shall I change that, oder is it okay as it is?

-due to the application of the automatic formatter some old stuff is shown as diff... its just another format, nothing more...

### CODE REVIEW

**Comment by vonnieda** [comment]
Hi @mgrl, I'm curious why you implemented this in ReferenceHead instead of ReferenceMachine? As this code is now, if one head succeeds but another fails isHomed will still be set true. I think this should be done in ReferenceMachine.home(). If that's okay with you I'll change and merge.

Thanks,
Jason

**Comment by mgrl** [comment]
Or maybe ishomed has to be stored to every head and ishomed in machine iterates over all heads?

**Comment by mgrl** [comment]
@vonnieda okay, just tested with the nulldriver:

the home() function in reference machine and
```
    @Override
    public void home() throws Exception {
        Logger.debug("home reference machine");
        super.home();
    }
```

its pendant in abstract machine never get called in the homing cycle.
```
    @Override
    public void home() throws Exception {
        Logger.debug("test");
        for (Head head : heads) {
            head.home();
        }
    }
```

is this a bug?

File: src/main/java/org/openpnp/gui/MachineControlsPanel.java [hunk 1]
```diff
@@ -272,7 +272,7 @@ public void actionPerformed(ActionEvent arg0) {
 					Configuration.get().getMachine().setEnabled(enable);
 					setEnabled(true);
 					if (machine.getHomeAfterEnabled() && machine.isEnabled()) {
-						selectedTool.getHead().home();
```
Comment (mgrl) [inline src/main/java/org/openpnp/gui/MachineControlsPanel.java]: okay, changed the gui elements to perform a homing cycle on the machine instead directly on the head. with the nulldriver it works as expected now.

**Comment by mgrl** [comment]
well, I found in the calling history that the gui directly calls .home on the head. I changed that in the latest commits. 
should be good then

**Comment by mgrl** [comment]
@vonnieda tested now on machine, works as intended (at least for 1 head - has anybody have two heads? he must have homed both heads separate by choosing a nozzle attached to that head...)

**Comment by vonnieda** [comment]
Great work, thank you @mgrl!

**Pull request merged**